src/Storages/StorageURL.cpp

```diff
@@ -347,7 +351,7 @@ Chunk StorageURLSource::generate()
     return {};
 }
 
-std::tuple<Poco::URI, std::unique_ptr<ReadWriteBufferFromHTTP>> StorageURLSource::getFirstAvailableURIAndReadBuffer(
+std::pair<Poco::URI, std::unique_ptr<ReadWriteBufferFromHTTP>> StorageURLSource::getFirstAvailableURIAndReadBuffer(
```

Comment:
Need to skip empty files in this function too. Otherwise `url('https://example.com/{empty|nonempty}.csv')` will return empty result (from `empty.csv`), and `url('https://example.com/{empty|nonempty}.parquet')` will fail, I think.

(If all files are empty, I guess return the last of them, which the caller will then immediately skip.)

Should be easy to add some `{x|y|z}` url globs in test_storage_s3 too.